addons/web_tour_recorder/static/tests/tour_recorder.test.js

```diff
@@ -0,0 +1,393 @@
+import { expect, test } from "@odoo/hoot";
+import { click, edit, keyDown, keyUp, queryAllTexts, queryOne } from "@odoo/hoot-dom";
+import { animationFrame, runAllTimers } from "@odoo/hoot-mock";
+import {
+    contains,
+    defineWebModels,
+    mountWithCleanup,
+    serverState,
+} from "@web/../tests/web_test_helpers";
+import { WebClient } from "@web/webclient/webclient";
+import { browser } from "@web/core/browser/browser";
+
+async function _checkSteps(steps) {
+    click(".o_button_steps");
+    await animationFrame();
+    expect(queryAllTexts(".o_tour_step_trigger")).toEqual(steps);
+    click(".o_button_steps");
+    await animationFrame();
+}
```

Comment:
Seems like you're still reading the steps from the UI instead of reading it from the state of the service/component. While not a huge problem it's definitely not a good practice and is pretty brittle, even if you have isolated the trigger to its own DOM element with a special class. For example, this function doesn't work if a test tries to use it while the tour steps are already expanded. You also need to wait for two animation frames every time. Here is an alternative suggestion:

```js
let tourRecorder;
beforeEach(() => {
    patchWithCleanup(TourRecorder.prototype, {
        setup() {
            tourRecorder = this;
            return super.setup(...arguments);
        },
    });
});
const checkTourSteps = (expected) => {
    expect(tourRecorder.state.steps.map(s => s.trigger)).toEqual(expected);
};
```